### DISCUSSION TITLE
resetting a controlled DatePicker

### DISCUSSION THREAD

**Discussion by Huiet** [mantinedev/mantine#3002] [Q&A]
I'm calling reset on submit for a datepicker, but when the user clicks to enter another date and clicks away (triggers a blur effectively) it will set the value to be the last selected date before the stateful value was cleared.

MVP code:
```
const [myDate, setMyDate] = useState(null);

const onSubmit = () => {//... do submit things;
setMyDate(null)
}

return (
 <DatePicker
        label={'Enter a  Date'}
        autoComplete={'off'}
        allowFreeInput
        value={myDate}
        onChange={(value) => setMyDate(value)}
/>

<button onClick={onSubmit}>submit<button>
)
```

**Comment by rtivital** [comment]
Do you have an issue with resetting date? It is not clear what the question is.

**Reply by Huiet** [reply]
effectively, when I reset the date it appears to reset, but if you were to click into/focus the calendar and then click away without selecting a date it would populate with the previously selected date

I would expect it to remain null.
